Alicerce/Sources/Network/Mappable.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Mappable.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 06/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public enum MappableError: Error {
+
+    case custom(String)
+
+    /// Unwrapps the enum custom value
+    var description: String {
+        switch self {
+        case let .custom(description): return description
+        }
+    }
+}
+
+public protocol Mappable {
+
+    /// Converts the object from a `Any` (Dictionary) to Itself.
+    ///
+    /// - Parameter object: 
+    /// An object that information to fill the model. Usually a Dictionary.
+    /// - Returns: 
+    /// The object itself filled with values from the object
+    /// - Throws: 
+    /// A MappableError
+    static func model(from object: Any) throws -> Self
+
+    /// Converts Itself into a `AnyObject` (Dictionary) representation
+    /// You can simply return `NSNull()` if it doesn't make sense in your context to do that.
+    ///
+    /// - Returns: 
+    /// An `AnyObject` (like a `Dictionary` representation)
+    func json() -> AnyObject
```

Comment:
Sorry, but i don't agree with make all computed properties if they don't have input parameters.

In the computer science terms, this makes sense to be a function because it can have logic assigned to it, it depends on the class that implements the method to decide what to do.

But this is just my point of view, i'm ok to change if it doesn't makes sense to you and the other guys.

But, let's not make all the `func` that doesn't have parameters as computed properties.